pkg/repo/fs.go

```diff
@@ -129,31 +128,31 @@ func (fsr *FsRepo) Open(c config.ReadWriter) error {
 		}
 	}
 
-	// modifies the config to include keys for accessing repo paths if they are not set.
-	// This ensures either user provided paths are valid to default paths for the repo are set.
-	fsr.EnsureRepoPathsConfigured(c)
-
-	cfg, err := c.Current()
-	if err != nil {
-		return err
+	// create files required by repo
+	if err := fsr.openRepoFiles(); err != nil {
```

Comment:
First, lets get rid of this package and move the logic under config. Maybe only migration logic should remain. Second, no files should be auto-created at an early level such as the repo or config. This will result in empty directories unnecessary created, such as executions dir created when you run a requester only node. Let `Node` type create the directories it needs only when trying to initialize a dependency

Reply:
Does this imply config is responsible for creating the necessary files and directories - e.g. `orchestrator_store`, `compute_store`, `user_id.pem`, and `compute_store/executions`? Or that the config should return constant values for these paths that are not configurable? 

I'd take issue with the former as I don't think the config should be making changes to the file system. If the latter, what service is responsible for ensuring the paths exist?

Reply:
As I explained, 'node' pkg should be the one initialzing and creating dependencies and the required paths. If it makes easier now during the migration to let config creat the dirs then go for it. No need for a separate repo pkg just for that purpose

Reply:
Do I understand the feedback correctly:
You want me to remove the methods I've added to the repo and instead add them to the config package?

Reply:
I want our code to only rely on config for node configuration. Having both config and repo for a subset of the configuration is confusing and unnecessary. For now, just have the path getters with lazy path creation in config, delete repo, and in the future we can have config read only and move path creation to node package

Reply:
You want the `pkg/repo` package deleted?